contrib/IECoreUSD/python/IECoreUSD/__init__.py

```diff
@@ -31,7 +31,19 @@
 #  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #
 ##########################################################################
-
-from _IECoreUSD import *
+import sys
+import platform
+import ctypes
 
 __import__( "IECore" ).loadConfig( "CORTEX_STARTUP_PATHS", subdirectory = "IECoreUSD" )
+
+# Load _IECoreUSD manually to ensure cross-platform compatibility
+sys = platform.system()
+if sys == "Darwin" :
+    libName = "libIECoreUSD.dylib"
+elif sys == "Windows" :
+    libName = "IECoreUSD.dll"
+else :
+    libName = "libIECoreUSD.so"
```

Comment:
Hmm, this is a bit tricky actually. You can use the [SConstruct `INSTALL_LIB_NAME` option](https://github.com/ImageEngine/cortex/blob/master/SConstruct#L599-L603) to install this under different names... At IE we install it as `libIECoreUSD-10.so` for example, but others could pick any name they want. So this init will need to be somehow dynamic, though I'm not sure I have a good suggestion. Maybe it gets installed via a `sed` command and we put `$INSTALL_LIB_NAME` in the file?

Reply:
Ah yes, that is a wrinkle. If we replace text in the `__init__.py` at build time, Is there any concern over this being the only source file with a string replacement token like this? 

I could also see creating a symlink on Linux / Mac in SConstruct that points `libIECoreUSD.so` -> `libIECoreUSD-10.so`. It would be familiar to Linux developers and keep this isolated to the build file. 

Windows doesn't _really_ support symlinks so it wouldn't work there, but I'm not sure how common this kind of library naming scheme is on Windows. In my experience it's typical to just use the library name without decoration, but I don't know if that's true everywhere.

Reply:
In the Gaffer SConstruct, we do replacements for every single python source file as a matter of course : https://github.com/GafferHQ/gaffer/blob/master/SConstruct#L1070-L1073. In practice it's only used in one or two files, but it doesn't do any harm. I think doing the same for the python files for IECoreUSD would be fine, and then later on we can extend it to other modules in Cortex if needed.

Reply:
Hmm, although with Cortex, we run the tests using the Python files in the source directory, so it's not quite as simple as only needing to substitute during installation. What a pain...

Reply:
Right, I had missed the sed transform there. Maybe we change the tests to be installed to the build directory and run them from there? Gaffer does that, correct? Or is that stubbornly chasing after a feature (USD on Windows at least) that could be accomplished with a targeted fix just for MSVC?

Reply:
I'd be in favour of out-of-tree builds in general, but I wouldn't want to bottleneck your windows PR on that... nor would I want to burden you with the task of updating the SConstruct for it...

Reply:
Yeah, that would bottleneck this PR somewhat. I'm up for taking on moving the tests to the build directory if it's blocking this PR though. Do you think it's worth that effort now, or is there a better way forward?